monkey/monkey_island/cc/resources/agent_events.py

```diff
@@ -68,16 +69,39 @@ def get(self):
 
     def _parse_event_filter_args(
         self,
-    ) -> Tuple[Optional[Type[AbstractAgentEvent]], Optional[bool], Optional[Tuple[str, float]]]:
+    ) -> Tuple[
+        Optional[Type[AbstractAgentEvent]],
+        Optional[str],
+        Optional[bool],
+        Optional[Tuple[str, float]],
+    ]:
         type_arg = request.args.get("type", None)
+        tag_arg = request.args.get("tag", None)
         success_arg = request.args.get("success", None)
         timestamp_arg = request.args.get("timestamp", None)
 
+        type_ = self._parse_type_arg(type_arg)
+        tag = self._parse_tag_arg(tag_arg)
+        success = self._parse_success_arg(success_arg)
+        timestamp_constraint = self._parse_timestamp_arg(timestamp_arg)
+
+        return type_, tag, success, timestamp_constraint
+
+    def _parse_type_arg(self, type_arg: Optional[str]) -> Optional[Type[AbstractAgentEvent]]:
         try:
             type_ = None if type_arg is None else self._agent_event_registry[type_arg]
         except KeyError:
-            raise Exception(f'Unknown agent event type "{type_}"')
+            raise Exception(f'Unknown agent event type "{type_arg}"')
+
+        return type_
+
+    def _parse_tag_arg(self, tag_arg: Optional[str]) -> Optional[str]:
+        if tag_arg and not re.match(pattern=re.compile(EVENT_TAG_REGEX), string=tag_arg):
+            raise Exception(f'Invalid event tag "{tag_arg}"')
```

Comment:
Errors should be specific.
```suggestion
            raise ValueError(f'Invalid event tag "{tag_arg}"')
```